solutions/silver/cf-1744F.mdx

```diff
@@ -0,0 +1,144 @@
+are looking for.
+
+We need to find all subsegments that contain $l \ldots r$, that don't contain
+$\texttt{pos}$. Say $\texttt{pos}>r$, since we only need to slightly change our approach for
+$\texttt{pos} < l$. We can increment $r$ from $r$ to $\texttt{pos}-1$ fast enough, since each time
+we increment we increase the size of our subsegment by $1$, for a maximum of $N$
+times.
+
+Considering this new bound of $r$, we can find the number of possible values of
+$l$ that form a valid subsegment. We know that the $\texttt{mex}$ is correct for this
+right bound of $r$ and all left bounds $\leq l$, since the original $l \ldots r$
+covered $0, 1, \ldots, \texttt{mex} - 1$, and $\texttt{pos} > r$. Therefore, we just need to find
+the number of\ positions $\leq l$ such that $r-l+1 \leq \texttt{mex} \cdot 2$. The lower
+bound of our smallest value of $l$ is $r-\texttt{mex} \cdot 2 + 1$, which can be found by
+rearranging the previous inequality. Note that if this value is negative, it
+must be set to $0$. After this, we know that all values between $l$ and this
+smallest $l$ are valid left bounds for our given $r$, so we can add
+$l-\texttt{smallest\_l} + 1$. Again, note that if $l < \texttt{smallest\_l}$, which will happen in
+the case that our bounds $l$ and $r$ are too big for our current $\texttt{mex}$, we should instead add $0$.
+
+We can do the same if $\texttt{pos} < l$, by decrementing from $l$ to $\texttt{pos}+1$ with minimal
+changes to the math.
+
+To quickly find the next position of $\texttt{mex}$, we can store the position of each integer.
+
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(N)$
+
+
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+
+using namespace std;
+
+using ll = long long;
+
+int main() {
+	int t;
+	cin >> t;
+	while (t--) {
+		int n;
+		cin >> n;
+
+		// stores position of integer
+		vector<int> pos(n);
+
+		for (int i = 0; i < n; i++) {
+			int x;
+			cin >> x;
+			pos[x] = i;
+		}
+
+		// initialize left and right bounds for mex
+		int l = pos[0], r = pos[0];
+		ll ans = 1;
+
+		// mex represents the current mex l and r cover
+		for (int mex = 1; mex < n; mex++) {
+			int next_pos = pos[mex];
+
+			// skip if next_pos is already between l and r
+			if (l <= next_pos && next_pos <= r) { continue; }
+
+			/*
+			 * Since we can create a valid subsegment for a given mex up until
+			 * size mex * 2, our max difference between l and r is mex * 2 - 1.
+			 */
+			int max_diff = mex * 2 - 1;
+
+			if (next_pos < l) {
+				for (; l > next_pos; l--) {
+					int largest_r = min(l + max_diff, n - 1);
+					// add all possible right bounds that don't change the
+					// current mex
+					ans += max(largest_r - r + 1, 0);
+				}
+			} else {
+				for (; r < next_pos; r++) {
+					int smallest_l = max(r - max_diff, 0);
+					// add all possible left bounds that don't change the
+					// current mex
+					ans += max(l - smallest_l + 1, 0);
+				}
+			}
+		}
+
+		cout << ans << endl;
```

Comment:
```suggestion
		cout << ans << '\n';
```
endl's in a loop are kinda pointless ngl
(optional though, even i used to do this)